src/compiler/semantic/rules.rs

```diff
@@ -251,10 +249,49 @@ fn redeclare_rule(
     }
 }
 
+// Cannot use true or false if it's not inside of a logic expression.
+fn true_false_rule(analyser: &mut Analyser, expr: &Expression<BigInt, Identifier>) {
+    use Expression::*;
+
+    match expr {
+        BinOp { lhs, rhs, .. } => {
+            check_true_false(analyser, expr, lhs);
+            check_true_false(analyser, expr, rhs);
```

Comment:
@rutefig I think that the recursion into true_false_rule is done inside check_true_false is confusing. It is easier to read if true_false_rule calls itself after calling check_true_false.

Reply:
```
BinOp { lhs, rhs, .. } => {
            check_true_false(analyser, expr, lhs);
            check_true_false(analyser, expr, rhs);
            true_false_rule(analyser, lhs);
            true_false_rule(analyser, rhs);
```